OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs

```diff
@@ -207,41 +207,45 @@ public SaveDirectory(Folder folder, string displayName, MapClassification classi
 					return;
 
 				var combinedPath = Platform.ResolvePath(Path.Combine(selectedDirectory.Folder.Name, filename.Text + fileTypes[fileType].Extension));
+				CheckOverwriteMap(modData, map, combinedPath, saveMap);
+			};
+		}
 
-				if (map.Package?.Name != combinedPath)
+		public static void CheckOverwriteMap(ModData modData, Map map, string combinedPath, Action<string> saveMap)
```

Comment:
Perhaps change the name to `SaveMap` or something like that. The name doesn't fit as this does not just check, but a save